aws/resource_aws_msk_cluster_test.go

```diff
@@ -0,0 +1,286 @@
+	enhanced_monitoring = "%s"
+}`, rInt, monitoring_type)
+}
+
+func testAccCheckMskClusterExists(n string, cluster *kafka.ClusterInfo) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No MSK Cluster ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).kafkaconn
+		describeOpts := &kafka.DescribeClusterInput{
+			ClusterArn: aws.String(rs.Primary.Attributes["arn"]),
+		}
+		resp, err := conn.DescribeCluster(describeOpts)
+		if err != nil {
+			return err
+		}
+
+		*cluster = *resp.ClusterInfo
+
+		return nil
+	}
+}
+
+func testAccCheckAWSMskClusterAttributes(cluster *kafka.ClusterInfo) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if !strings.HasPrefix(*cluster.ClusterName, "terraform-msk-test") {
+			return fmt.Errorf("Bad Cluster name: %s", *cluster.ClusterName)
+		}
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != "aws_msk_cluster" {
+				continue
+			}
+			if *cluster.ClusterArn != rs.Primary.Attributes["arn"] {
+				return fmt.Errorf("Bad Cluster ARN\n\t expected: %s\n\tgot: %s\n", rs.Primary.Attributes["arn"], *cluster.ClusterArn)
+			}
+			brokerCount := strconv.Itoa(int(aws.Int64Value(cluster.NumberOfBrokerNodes)))
+			if brokerCount != rs.Primary.Attributes["broker_count"] {
+				return fmt.Errorf("Bad Cluster Broker Count\n\t expected: %s\n\tgot: %s\n", rs.Primary.Attributes["broker_count"], brokerCount)
+			}
+			volumeSize := strconv.Itoa(int(aws.Int64Value(cluster.BrokerNodeGroupInfo.StorageInfo.EbsStorageInfo.VolumeSize)))
+			if volumeSize != rs.Primary.Attributes["broker_volume_size"] {
+				return fmt.Errorf("Bad Broker Volume Size\n\t expected: %s\n\tgot: %s\n", rs.Primary.Attributes["broker_volume_size"], volumeSize)
+			}
+			encryptRestKey := *cluster.EncryptionInfo.EncryptionAtRest.DataVolumeKMSKeyId
+			if !strings.Contains(encryptRestKey, rs.Primary.Attributes["encrypt_rest_key"]) {
+				return fmt.Errorf("Bad Encrypt Rest Key\n\t expected: %s\n\tgot: %s\n", rs.Primary.Attributes["encrypt_rest_key"], encryptRestKey)
+			}
+			if *cluster.ZookeeperConnectString == "" {
+				return fmt.Errorf("empty zookeeper_connect")
+			}
+		}
+		return nil
+	}
+}
+
+func testAccCheckMskClusterDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_msk_cluster" {
+			continue
+		}
+		conn := testAccProvider.Meta().(*AWSClient).kafkaconn
+		describeOpts := &kafka.DescribeClusterInput{
+			ClusterArn: aws.String(rs.Primary.Attributes["arn"]),
+		}
+		resp, err := conn.DescribeCluster(describeOpts)
+		if err == nil {
+			if resp.ClusterInfo != nil && *resp.ClusterInfo.State != "DELETING" {
+				return fmt.Errorf("Error: Cluster still exists")
+			}
+		}
+
+		return nil
```

Comment:
This should be removed to allow the `for` loop to check all `aws_msk_cluster` resources in the state